core/src/main/kotlin/me/lazmaid/kraph/lang/Variables.kt

```diff
@@ -1,10 +1,22 @@
 package me.lazmaid.kraph.lang
 
+import me.lazmaid.kraph.KraphVariable
+
 /**
  * Created by OinkIguana on 10/25/2017 AD.
  */
 
-// TODO: implement variables
-internal class Variables() {
-    fun print(): String? = null
+internal class Variables {
+
+    val variables = mutableMapOf<String, KraphVariable>()
+
+    fun print(): String? = variables
+            .takeIf { it.isNotEmpty() }
+            ?.let {
+                it.entries.joinToString(separator = ",", prefix = "{", postfix = "}") { "\"${it.key}\": ${it.value.jsonValue}" }
+            }
+
+    fun asArgument(): Argument? =
```

Comment:
IMO, this should be an extension of this class more than a member function itself.

Reply:
I don't really care.
It would make some classes (such as Variables) not internal - do you mind?

Reply:
I think there is no different in this case. You may leave it like this.